### PR TITLE
[#152241791] Add documentation for private Docker

### PR DESCRIPTION
**Pull request by samcrang**
## What

We've added support for private Docker registries—this documents it

## How to review

- read it, make sure it makes sense
- compare it to the Google doc linked to in the PT story

## Who can review

Anyone but @samcrang or @jonathanglassman 

## Before merge

Squash commits where it makes sense to keep commit history clean

### CODE REVIEW

File: source/documentation/guidance/buildpacks.md [hunk 1]
```diff
@@ -31,18 +31,17 @@ Regularly maintain your buildpack to update your runtime and dependencies as new
...
 
 ### Custom Buildpacks compared to Docker Images
 
 Custom buildpacks provide some advantages over choosing a Docker solution:
 
-- they are not dependent on [Docker Hub](https://hub.docker.com/) availability
-a single way to specify app dependencies
-- large variety of open source buildpacks, meaning your deployment pipeline doesn’t need to include docker image creation
-- it is designed for 12-factor; some Docker images may contain databases or other forms of storage that should be provisioned as an external backing service
+- they are not dependent on Docker registry such as [Docker Hub](https://hub.docker.com/) being available
```
Comment (bandesz) [inline source/documentation/guidance/buildpacks.md]: this should say ```on a Docker registry``` (adding "a")

File: source/documentation/overview/benefits.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Features of the PaaS platform currently include:
 
 *   backing service support for PostgreSQL, MySQL and MongoDB
 *   language support through the [standard Cloud Foundry buildpacks](http://docs.cloudfoundry.org/buildpacks/) [external link] except .NET Core
-*   limited support for custom buildpacks and Docker images
+*   support for public and private Docker images
```
Comment (bandesz) [inline source/documentation/overview/benefits.md]: ```limited support for custom buildpacks``` was removed by mistake (separate bullet point)

File: source/documentation/guidance/buildpacks.md [hunk 3]
```diff
@@ -31,18 +31,17 @@ Regularly maintain your buildpack to update your runtime and dependencies as new
 
 #### Docker Images
 
-You are responsible for your docker container and custom image. Learn about this [experimental feature](https://docs.cloud.service.gov.uk/#deploy-a-docker-image-experimental).
+You are responsible for your Docker container and custom image. Learn about this [experimental feature](https://docs.cloud.service.gov.uk/#deploy-a-docker-image-experimental).
 
 ### Custom Buildpacks compared to Docker Images
```
Comment (bandesz) [inline source/documentation/guidance/buildpacks.md]: this should be: ```Custom buildpacks compared to Docker images``` (B -> b, I -> i)

File: source/documentation/guidance/buildpacks.md [hunk 4]
```diff
@@ -31,18 +31,17 @@ Regularly maintain your buildpack to update your runtime and dependencies as new
 
 #### Docker Images
```
Comment (bandesz) [inline source/documentation/guidance/buildpacks.md]: This should be ```Docker images``` (I -> i)

Comment (samcrang) [reply source/documentation/guidance/buildpacks.md]: Yeah, I've mentioned this to @jonathanglassman already; he's going to fix up these things separately.

**Comment by samcrang** [comment]
I've fixed up the broken grammar and missing bullet-points. On the sub-heading capitalisation weirdness @jonathanglassman knows about it and he's going to fix that up separately.

**Comment by bandesz** [comment]
While reviewing this a found this in the docs: 

```If your language is not supported by standard buildpacks, we recommend that for a community supported custom buildpack that is maintained and updated by that community.```

Do we want to fix this in this PR? Probably it should say:

```If your language is not supported by standard buildpacks, we recommend to use a community supported custom buildpack that is maintained and updated regularly.```

**Comment by samcrang** [comment]
I feel like there's probably other stuff in the tech docs that need to change. I'm not sure it's really in-scope of the PR. Even this change feels dubious, do we actually _recommend_ people use community supported buildpacks? Or is it more of a "it is possible to use them" thing?

**Comment by bandesz** [comment]
@samcrang let's leave this one out, I just didn't want it to be lost

**Pull request merged**